#!/bin/sh /usr/share/dpatch/dpatch-run
## 05_sftpdriver.c_fix-for-new-openssh.dpatch by Agustin Martin Domingo <agmartin@debian.org>
##
## DP: Patch for sftp protocol doesn't work with >= 1:4.2p1-1
## the new sftp command doesn't print a prompt after connecting to the remote
## server any longer if STDIN is not a tty.[1]
## [1] http://www.openbsd.org/cgi-bin/cvsweb/src/usr.bin/ssh/sftp.c.diff?r1=1.65&r2=1.66&f=h

@DPATCH@
diff -urNad sitecopy-0.16.3~/src/sftpdriver.c sitecopy-0.16.3/src/sftpdriver.c
--- sitecopy-0.16.3~/src/sftpdriver.c	2005-06-29 11:44:47.000000000 +0200
+++ sitecopy-0.16.3/src/sftpdriver.c	2007-10-22 14:58:51.000000000 +0200
@@ -140,7 +140,6 @@
     }
     close(c_in);
     close(c_out);
-    read_sftp(sess);    /* wait for prompt */
     return SITE_OK;
 }

